src/state.ts

```diff
@@ -140,14 +141,20 @@ export async function initProjectDir(uri?: vscode.Uri): Promise<void> {
     }
 }
 
-async function findLocalProjectRoot(projectFileNames, doc, workspaceFolder): Promise<void> {
+function isNamespace(path: string): boolean {
+    const content = filesCache.content(path);
+    const hasNsDecl = /^\(ns /.test(content);
+    return hasNsDecl;
+}
```

Comment:
This is a bit brittle. People have white space and comments and whatever before the ns form. I think that if we refactor out a version of `src/namespace.ts:getNameSpace()` that takes a string as its argument, that'll be more safer and more battle tested. I can help with this if you like. I think I might have created a thing that can be used in the mega-parinfer PR: https://github.com/BetterThanTomorrow/calva/blob/6e756d7d844268f0c5dfabd49e3481f0a91bd1a7/src/cursor-doc/model.ts#L489

Reply:
And no that PR is merged!